plugins/typescript/src/generators/generateReactQueryComponents.ts

```diff
@@ -183,6 +214,16 @@ export const generateReactQueryComponents = async (
     }
   );
 
+  const queryKeyManager = f.createTypeAliasDeclaration(
+    undefined,
+    [f.createModifier(ts.SyntaxKind.ExportKeyword)],
+    "QueryOperation",
+    undefined,
+    keyManagerItems.length > 0
+      ? f.createUnionTypeNode(keyManagerItems)
+      : f.createKeywordTypeNode(SyntaxKind.NeverKeyword)
```

Comment:
![image](https://user-images.githubusercontent.com/1761469/159034899-438f9922-a9f7-4392-9a7b-e6386bc9d32c.png)

This will break the `queryKeyFn` types if the value is `never`

Reply:
![image](https://user-images.githubusercontent.com/1761469/159035533-9e535778-ea2e-49e8-8738-1925ae41b7ef.png)

This type works as fallback